### DISCUSSION TITLE
Enhanced exercise specific day once a month?

### DISCUSSION THREAD

**Discussion by biggen1684** [jgyates/genmon#1143] [General]
The way I read the enhanced exercise option, there is no way to set a specific day of the week when set for once a month exercise. For example, if I want to run an enhanced loaded exercise on the 1st Monday of the month, there is no way to configure this. I have to pick a date only. Is this correct?

**Comment by jgyates** [comment]
you are correct. The monthly option only allows you select the day of the month (i.e. the 1st or the 27th, etc.)

**Reply by biggen1684** [reply]
Thanks for the quick response! Ah ok. Is it worth creating a pull request to ask to make it a specific day when on monthly? Has anyone else asked?

Thanks for this awesome program by the way. Really a great piece of kit!

**Reply by jgyates** [reply]
So far no one else has asked. The change to the add on would be straightforward. Displaying the option in the UI without confusion with the existing add on javascript code is probably more time consuming. The current code does not dynamically change the options based on the type of exercise selected so unless you changed that functionality you would end up with entry boxes for both  ways to set the monthly exercise day.

**Reply by biggen1684** [reply]
You want me to create an issue for it for tracking? You can, of course, choose whether it’s worth your time or not to add it.

**Reply by jgyates** [reply]
This thread is fine for tracking. I will add it to the feature request list. I will think on this. I may make this a feature, but only enabled by editing the /etc/genmon/genexercise.conf file. This path would allow the fuctionality but not the extra work or clutter of the exiting UI.

**Reply by jgyates** [reply]
https://github.com/jgyates/genmon/blob/master/feature_requests.md

**Reply by biggen1684** [reply]
Thank you so much even for the consideration! I appreciate that.

**Reply by jgyates** [reply]
I checked in an update that should allow you to specify the xth week day of every month. The genexercise.conf file in the repository has info on the mods to the file /etc/genmon/genexercise.conf file that need to made on your system. To enable this:

1) Modify the file /etc/genmon/genexercise.conf file adding the following line:

     exercise_nth_day_of_month = 1

If you want the 1st Saturday then set exercise_nth_day_of_month to 2 and set exercise_day_of_week to Saturday

2) Make sure the Enhanced exercise add on is enabled on the add on page
3) update the repository from the About page 

Let me know if you have any issues or questions.

**Reply by biggen1684** [reply]
Wow! Thats gotta be the fastest commit from request ever made. Thank you so much! I’ll go ahead and give it a go and report back any issues.
